src/librustc_attr/builtin.rs

```diff
@@ -908,11 +908,12 @@ pub fn find_repr_attrs(sess: &ParseSess, attr: &Attribute) -> Vec<ReprAttr> {
                     let parse_alignment = |node: &ast::LitKind| -> Result<u32, &'static str> {
                         if let ast::LitKind::Int(literal, ast::LitIntType::Unsuffixed) = node {
                             if literal.is_power_of_two() {
-                                // rustc::ty::layout::Align restricts align to <= 2^29
-                                if *literal <= 1 << 29 {
+                                // Many targets don't support global variables
+                                // with alignment greater than the page size.
+                                if *literal <= 4096 {
                                     Ok(*literal as u32)
                                 } else {
-                                    Err("larger than 2^29")
+                                    Err("larger than 4096")
```

Comment:
if we land this, then it would be good to mention what this magic number is about